classes/category/Repository.php

```diff
@@ -63,32 +71,27 @@ public function exists(int $id, ?int $contextId = null): bool
         return $this->dao->exists($id, $contextId);
     }
 
-    /**
-     * Get the breadcrumb of a category
-     *
-     * @return string For example: Social Sciences > Anthropology
-     */
-    public function getBreadcrumb(Category $category, ?Category $parent = null): string
-    {
-        return !$parent
-            ? $category->getLocalizedTitle()
-            : __('common.categorySeparator', [
-                'parent' => $parent->getLocalizedTitle(),
-                'child' => $category->getLocalizedTitle()
-            ]);
-    }
-
     /**
      * Get the breadcrumbs for a Collection of categories
      */
     public function getBreadcrumbs(LazyCollection $categories): LazyCollection
     {
         return $categories->map(function (Category $category) use ($categories) {
-            /** @var ?Category $parent */
-            $parent = $categories->first(
-                fn (Category $c) => $c->getId() === $category->getParentId()
-            );
-            return $this->getBreadcrumb($category, $parent);
+            $currentCategory = $category;
+            $result = $currentCategory->getLocalizedTitle();
+
+            // Traverse up category tree until we reach a top-level category
+            while ($currentCategory->getParentId() && $parent = $categories->get($currentCategory->getParentId())) {
```

Comment:
Such loops can be dangerous, it's better to protect against circular references.

p.s.: I'm not a big fan of building trees with `parentId`, there are better ways :)